packages/compiler-cli/src/ngtsc/scope/src/local.ts

```diff
@@ -390,9 +417,11 @@ export class LocalModuleScopeRegistry implements MetadataRegistry, ComponentScop
         exportPipes.set(decl.node, pipe);
       } else {
         // decl is an unknown export.
-        if (this.localReader.getDirectiveMetadata(decl) !== null ||
-            this.localReader.getPipeMetadata(decl) !== null) {
-          diagnostics.push(invalidReexport(decl, ngModule.rawExports));
+        const dirMeta = this.fullReader.getDirectiveMetadata(decl);
+        const pipeMeta = this.fullReader.getPipeMetadata(decl);
+        if (dirMeta !== null || pipeMeta !== null) {
+          const isStandalone = (dirMeta?.isStandalone || pipeMeta?.isStandalone)!;
```

Comment:
nit: a `??` would be slightly more appropriate here, also allowing to remove one `?.` in favor of `!`.

```suggestion
          const isStandalone = dirMeta?.isStandalone ?? pipeMeta!.isStandalone;
```

Reply:
I think this is too magical actually (both the original and the suggested) - I've changed it to be more explicit.